src/ext-test/java/org/opentripplanner/ext/legacygraphqlapi/GraphQLApiTest.java

```diff
@@ -0,0 +1,62 @@
+package org.opentripplanner.ext.legacygraphqlapi;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.opentripplanner.test.support.JsonAssertions.assertEqualJson;
+
+import jakarta.ws.rs.core.Response;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import org.glassfish.jersey.message.internal.OutboundJaxrsResponse;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.opentripplanner.standalone.api.OtpServerRequestContext;
+import org.opentripplanner.test.support.FilePatternSource;
+import org.opentripplanner.test.support.PrettyPrinter;
+
+@Execution(ExecutionMode.CONCURRENT)
+class GraphQLApiTest {
+
+  static OtpServerRequestContext context = new TestServerRequestContext();
+  static LegacyGraphQLAPI resource = new LegacyGraphQLAPI(context, "ignored");
+
+  @FilePatternSource(pattern = "src/ext-test/resources/legacygraphqlapi/queries/*.graphql")
+  @ParameterizedTest(name = "Check GraphQL query in {0}")
+  void graphQL(Path path) throws IOException {
+    var query = Files.readString(path);
+    var response = resource.getGraphQL(query, 2000, 10000, new TestHeaders());
+    var actualJson = extracted(response);
+    assertEquals(200, response.getStatus());
+
+    var expectationFile = path
+      .getParent()
+      .getParent()
+      .resolve("expectations")
+      .resolve(path.getFileName().toString().replace(".graphql", ".json"));
```

Comment:
can be extracted into a private method for easier reading of the test - also add a short JavaDoc comment to the method for the rule for the relationship between dir `expectations` and `queries `. For example something like:
```
/**
 * Locate 'expectations' relative to the given query input file. The 'expectations' and 'queries' directories 
 * are expected to be in the same directory.
 */
private static Optional<Path> findExpectatiosFilePathRelativeToQueryFile(Path qureyfile) {...} 
```

Reply:
Done in https://github.com/opentripplanner/OpenTripPlanner/pull/4706/commits/ee4d828ef8c59bf92e488d753ef25018fa77ba6e

I can't believe I'm saying this but it felt more natural to _not_ use an Optional as File is already kind of optional as it has a `exitsts` method.

Reply:
Ok with me